File: scripts/fetch_logs.js
import { CloudWatchLogsClient, FilterLogEventsCommand } from "@aws-sdk/client-cloudwatch-logs";
import { fromIni } from "@aws-sdk/credential-providers";

async function getLogs() {
  const client = new CloudWatchLogsClient({ 
    region: "us-east-1", 
    credentials: fromIni({profile: "Matthew_Cli"}) 
  });
  const params = {
    logGroupName: "/aws/lambda/CertPrep360-Dev-GetQuestions",
    startTime: Date.now() - 30 * 60 * 1000, // last 30 minutes
  };
  try {
    const command = new FilterLogEventsCommand(params);
    const data = await client.send(command);
    data.events.slice(-20).forEach((event) => {
      console.log(event.message);
    });
  } catch (error) {
    console.error("Error fetching logs:", error);
  }
}
getLogs();

File: scripts/fetch_logs.py
import boto3
import time
from datetime import datetime, timedelta

def main():
    client = boto3.client('logs', region_name='us-east-1')
    group_name = '/aws/lambda/CertPrep360-Dev-GetQuestions'
    
    start_time = int((datetime.now() - timedelta(hours=2)).timestamp() * 1000)
    
    try:
        response = client.filter_log_events(
            logGroupName=group_name,
            startTime=start_time,
            filterPattern='ERROR'
        )
        events = response.get('events', [])
        print(f"Found {len(events)} Error events.")
        for e in events[-20:]:  # print last 20
            print(e['message'].strip())
            
    except Exception as e:
        print(f"Boto3 error: {e}")

if __name__ == '__main__':
    main()
